mesonbuild/backend/ninjabackend.py

```diff
@@ -3575,7 +3600,12 @@ def generate_ending(self):
             for t in deps.values():
                 # Add the first output of each target to the 'all' target so that
                 # they are all built
-                targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
+                #Add archive file if shared library in AIX for build all.
+                if mesonlib.is_aix():
+                    aix_tmp = re.sub ('[.][a]([.]?([0-9]+))*([.]?([a-z]+))*','.a', t.get_outputs()[0].replace ('.so','.a'))
+                    targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0] if mesonlib.is_aix() and not(t.get_outputs()[0].endswith('.so')) else aix_tmp))
+                else:
+                    targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
```

Comment:
Similar comments. Should this maybe be extracted into a helper function?

Reply:
So this is actually smaller now. Kindly check. There was no need to again check the target machine as @dcbaker mentioned